### PR TITLE
Logging improvement 

### PR DESCRIPTION
**Pull request by foram-splunk**
Added monitoring module for metrics collection.The following metrics are getting collected : 

- nozzle.queue.percentage (Shows how much internal queue is filled)
- splunk.events.dropped.count (Number of events dropped from splunk HEC)
- splunk.events.sent.count (Number of events sent to splunk)
- firehose.events.dropped.count (Number of events dropped from nozzle)
- firehose.events.received.count (Number of events recieved from firehose(websocket))
- splunk.events.throughput (Average Payload size)
- nozzle.usage.ram  (RAM Usage)
- nozzle.usage.cpu (CPU Usage)
- nozzle.cache.memory.hit (Number of times it has successfully retrieved the data from memory)
- nozzle.cache.memory.miss (Number of times it has unsuccessfully tried to retrieved the data from memory)
- nozzle.cache.remote.hit (Number of times it has successfully retrieved the data from remote)
- nozzle.cache.remote.miss (Number of times it has unsuccessfully tried to retrieved the data from remote)
- nozzle.cache.boltdb.hit (Number of times it has successfully retrieved the data from BoltDB)
- nozzle.cache.boltdb.miss (Number of times it has unsuccessfully tried to retrieved the data from BoltDB)

### CODE REVIEW

**Comment by harshit-splunk** [comment]
@foram-splunk please rebase with develop branch

**Review by harshit-splunk — APPROVED** [review]
LGTM!

File: README.md [hunk 1]
```diff
@@ -95,9 +95,12 @@ This is recommended for dev environments only.
...
-* `STATUS_MONITOR_INTERVAL`: Time interval (in s/m/h. For example, 3600s or 60m or 1h) for monitoring memory queue pressure. Use to help with back-pressure insights. (Increases CPU load. Use for insights purposes only) Default is 0s (Disabled).
-* `DROP_WARN_THRESHOLD`: Threshold for the count of dropped events in case the downstream is slow. Based on the threshold, the errors will be logged.
 * `SPLUNK_LOGGING_INDEX`: The Splunk index where logs from the nozzle of the sourcetype `cf:splunknozzle` will be sent to. Warning: Setting an invalid index will cause events to be lost. This index must match one of the selected indexes for the Splunk HTTP event collector token used for the SPLUNK_TOKEN parameter. When not provided, all logging events will be forwarded to the default SPLUNK_INDEX. The default value is `""`
+* `STATUS_MONITOR_INTERVAL`: Time interval (in s/m/h. For example, 3600s or 60m or 1h) for Enabling Monitoring(Metric data of insights with in the connectors). Default is 0s (Disabled).
```
Comment (omrozowicz-splunk) [inline README.md:99]: There's no space after `Monitoring` in `Monitoring(Metric`

File: README.md [hunk 2]
```diff
@@ -263,6 +266,31 @@ FORMAT = new_index
...
+|  `nozzle.queue.percentage` | Shows how much internal queue is filled
+|  `splunk.events.dropped.count` | Number of events dropped from splunk HEC
+|  `splunk.events.sent.count` | Number of events sent to splunk
+|  `firehose.events.dropped.count` | Number of events dropped from nozzle
+|  `firehose.events.received.count` | Number of events recieved from firehose(websocket)
+|  `splunk.events.throughput` | Average Payload size
+|  `nozzle.usage.ram` | RAM Usage 
+|  `nozzle.usage.cpu` | CPU Usage
+|  `nozzle.cache.memory.hit` | How many times it has successfully retrieved the data from memory
+|  `nozzle.cache.memory.miss` | How many times it has unsuccessfully tried to retrieved the data from memory
```
Comment (omrozowicz-splunk) [inline README.md:282]: I think it should be `tried to retreive`

**Review by omrozowicz-splunk — APPROVED** [review]

**Pull request merged**